search/search-client/src/main/java/io/camunda/search/clients/query/SearchIdsQuery.java

```diff
@@ -7,37 +7,30 @@
  */
 package io.camunda.search.clients.query;
 
-import static io.camunda.util.CollectionUtil.addValuesToList;
-import static io.camunda.util.CollectionUtil.collectValues;
-
 import io.camunda.util.ObjectBuilder;
-import java.util.Collections;
+import java.util.ArrayList;
+import java.util.Arrays;
 import java.util.List;
 import java.util.Objects;
-import java.util.function.Function;
-
-public final record SearchIdsQuery(List<String> values) implements SearchQueryOption {
 
-  static SearchIdsQuery of(final Function<Builder, ObjectBuilder<SearchIdsQuery>> fn) {
-    return SearchQueryBuilders.ids(fn);
-  }
+public record SearchIdsQuery(List<String> values) implements SearchQueryOption {
 
   public static final class Builder implements ObjectBuilder<SearchIdsQuery> {
 
-    private List<String> ids;
+    private final List<String> ids = new ArrayList<>();
 
     public Builder values(final List<String> values) {
-      ids = addValuesToList(ids, values);
+      ids.addAll(Objects.requireNonNullElse(values, List.of()));
```

Comment:
I know this has already been merged, but I still want to leave my feedback here.

IMO, this handling should be "centralized" in a static method or similar, as there are multiple cases where a list of values can be provided. That way, we would have one way of handling these cases and would (to some degree) prevent others from implementing it differently.

For example, we could adjust the `#addValueToList()` to the following:

```
  public static <T> List<T> addValuesToList(final List<T> list, final List<T> values) {
    final List<T> result = Objects.requireNonNullElse(list, List.of());
    result.addAll(values);
    return result;
  }
```

Which could be used that way here:

```
public Builder values(final List<String> values) {
  ids = addValuesToList(ids, values);
}
```

Reply:
Agree. I think when i first wrote it i actually wanted to hear your thoughts first about it and also regarding the other method

Reply:
@romansmirnov something like this https://github.com/camunda/camunda/pull/19736/commits/1cc6052c2bc4b219c9d24fce31283bf59a31f9a4 ? 🤔